src/app/main/main.controller.js

```diff
@@ -201,6 +201,61 @@
             }
             return true;
         };
+      
+      // Custom Widget Code Starts here
+        
+        vm.openCustomWidgetModal = function(){
+          
+            var customWidgetModal = {
+                backdrop: true,
+                keyboard: true,
+                modalFade: true,
+                scope: $rootScope,
+                templateUrl: 'app/components/modal/customWidgetModal.html',
+                controller: 'MainCtrl',
+                resolve: {
+                  metricsMetadata: function() {
+                    return $rootScope.metricsMetadata
```

Comment:
Would probably create a separate function, separate from `metricsMetadata`. To keep a list of metrics for the widget.